packages/backend-core/src/featureFlags/index.ts

```diff
@@ -36,18 +36,33 @@ export function isEnabled(featureFlag: string) {
 }
 
 export function getTenantFeatureFlags(tenantId: string) {
-  const flags = []
+  let flags: string[] = []
 
   if (TENANT_FEATURE_FLAGS) {
     const globalFlags = TENANT_FEATURE_FLAGS["*"]
-    const tenantFlags = TENANT_FEATURE_FLAGS[tenantId]
+    const tenantFlags = TENANT_FEATURE_FLAGS[tenantId] || []
+
+    // Explicitly exclude tenants from global features if required.
+    // Prefix the tenant flag with '!'
+    const tenantOverrides = tenantFlags.reduce((acc: string[], flag) => {
+      if (flag.startsWith("!")) {
+        let stripped = flag.substring(1)
+        acc.push(stripped)
+      }
+      return acc
+    }, [])
 
     if (globalFlags) {
       flags.push(...globalFlags)
     }
-    if (tenantFlags) {
+    if (tenantFlags.length) {
       flags.push(...tenantFlags)
     }
+
+    // Purge any tenant specific overrides
+    flags = flags.filter(flag => {
+      return tenantOverrides.indexOf(flag) == -1 && !flag.startsWith("!")
+    })
```

Comment:
Can we add a unit test for this? We really should have had one already. Now that we're adding more logic it would be a good time to start this, as it's a very critical flow. 

It'd be nice to test these variants for the value of `TENANT_FEATURE_FLAGS`

- `empty`
   - `flag1` isn't enabled for for `tenant1` 
- `tenant1:flag1`
   - `flag1` is enabled for `tenant1`
   - `flag1` is disabled for `tenant2`
- `tenant1:flag1,tenant1:flag2,tenant2:flag2`
   - `flag1` is enabled for `tenant1` 
   - `flag2` is enabled for `tenant1` 
   - `flag1` is disabled for `tenant2`
   - `flag2` is enabled for `tenant2`
- `*:flag1,tenant1:flag2`
   - `flag1` is enabled for tenant1
   - `flag1` is enabled for tenant2
   - `flag2` is enabled for tenant1
- `*:flag1,tenant2:!flag1`
   - `flag1` is enabled for tenant1
   - `flag1` is disabled for tenant2